src/bin/background-migration/index.ts

```diff
@@ -0,0 +1,194 @@
+          migrateRepo(repoId)
+            .then(async res => {
+              const status = res.status
+              const body = await res.json()
+
+              if (status < 200 || status >= 300) {
+                console.error(`${repoId} ${status} response:`, body)
+                await logFailedRepo(repoId, { status, body })
+              } else {
+                console.log(
+                  `Migrated ${repoId} (partition ${partitionHex} at index ${
+                    index + j
+                  })`
+                )
+              }
+            })
+            .catch(async (error: Error) => {
+              console.error(`${repoId} failed:`, error)
+              await logFailedRepo(repoId, {
+                error: { message: error.message, stack: error.stack }
+              })
+            })
+        )
+      )
+
+      index += concurrency
+
+      checkpoint = await updateCheckpoint({
+        partition: checkpoint.partition,
+        index
+      })
+    }
+
+    const nextPartition: number = checkpoint.partition + 1
+    if (nextPartition > 255) {
+      checkpoint = await updateCheckpoint({
+        partition: nextPartition,
+        index: 0
+      })
+    } else {
+      checkpoint = null
+    }
+  }
+}
+
+main().catch(criticalError)
+
+function criticalError(err: any): void {
+  console.error(err)
+  process.exit(1)
+}
+
+async function setup(): Promise<void> {
+  await mkdirp(dataDir)
+}
+
+// ---------------------------------------------------------------------
+// Lib
+// ---------------------------------------------------------------------
+
+async function getCheckpoint(): Promise<Checkpoint> {
+  const checkpointFileName = 'checkpoint.json'
+  const checkpointFilePath = join(dataDir, checkpointFileName)
+
+  const checkpoint = await getCleanFile(
+    checkpointFilePath,
+    asCheckpoint,
+    async () => {
+      return { partition: 0, index: 0 }
+    }
+  )
+
+  return checkpoint
+}
+
+async function updateCheckpoint(checkpoint: Checkpoint): Promise<Checkpoint> {
+  const checkpointFileName = 'checkpoint.json'
+  const checkpointFilePath = join(dataDir, checkpointFileName)
+
+  await writeFile(checkpointFilePath, JSON.stringify(checkpoint), 'utf8')
+
+  return checkpoint
+}
+
+async function getScanFile(partitionHex: string): Promise<ScanFile> {
+  const scanFileName = `${partitionHex}.json`
+  const scanFilePath = join(dataDir, scanFileName)
+
+  const scan = await getCleanFile(scanFilePath, asScanFile, () =>
+    scanRemotesForScanFile(partitionHex)
+  )
+
+  return scan
+}
+
+async function scanRemotesForScanFile(partitionHex: string): Promise<ScanFile> {
+  const repoIdArrays = await Promise.all(
+    sshHosts.map(sshHost =>
+      lsRemote(sshHost, join(remoteReposDir, partitionHex))
+    )
+  )
+  const repoIds = mergeArrays(...repoIdArrays)
+
+  return { partitionHex, repoIds }
+}
+
+function mergeArrays<T>(...arrays: T[][]): T[] {
+  return Array.from(new Set(arrays.flatMap(array => array)))
```

Comment:
Can't this just be

`return Array.from(new Set(arrays.flat())`

Reply:
Good idea.